packages/sdk/src/modules/JsonRpcDatasource.ts

```diff
@@ -0,0 +1,240 @@
+import { Transaction as BTCTransaction } from "bitcoinjs-lib";
+import { rpc } from "../api/jsonrpc";
+import { OrditSDKError } from "../errors";
+import { BaseDatasource } from "./BaseDatasource";
+import { DatasourceUtility } from "./DatasourceUtility";
+import type {
+  GetBalanceOptions,
+  GetInscriptionOptions,
+  GetInscriptionsOptions,
+  GetInscriptionUTXOOptions,
+  GetSpendablesOptions,
+  GetTxOptions,
+  GetUnspentsOptions,
+  GetUnspentsResponse,
+  RelayOptions,
+} from "../api/types";
+import type { Network } from "../config/types";
+import type { Transaction, UTXO, UTXOLimited } from "../transactions/types";
+import type { Inscription } from "../inscription/types";
+import type { JsonRpcPagination } from "./types";
+
+interface JsonRpcDatasourceOptions {
+  network: Network;
+}
+
+class JsonRpcDatasource extends BaseDatasource {
+  constructor({ network }: JsonRpcDatasourceOptions) {
+    super({ network });
+  }
+
+  async getBalance({ address }: GetBalanceOptions) {
+    if (!address) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    return rpc[this.network].call<number>(
+      "Address.GetBalance",
+      { address },
+      rpc.id,
+    );
+  }
+
+  async getInscription({ id, decodeMetadata }: GetInscriptionOptions) {
+    if (!id) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    id = id.includes(":")
+      ? id.replace(":", "i")
+      : !id.includes("i")
+      ? `${id}i0`
+      : id;
+
+    let inscription = await rpc[this.network].call<Inscription>(
+      "Ordinals.GetInscription",
+      { id },
+      rpc.id,
+    );
+    if (decodeMetadata) {
+      inscription = DatasourceUtility.transformInscriptions([inscription])[0];
+    }
+
+    return inscription;
+  }
+
+  async getInscriptionUTXO({ id }: GetInscriptionUTXOOptions) {
+    if (!id) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    id = id.includes(":")
+      ? id.replace(":", "i")
+      : !id.includes("i")
+      ? `${id}i0`
+      : id;
```

Comment:
@iamcrazycoder I will refactor this into a helper function.

Reply:
You can move this to utils.